pkg/controller/registry/reconciler/grpc.go

```diff
@@ -478,3 +495,29 @@ func (c *GrpcRegistryReconciler) podFailed(pod *corev1.Pod) (bool, error) {
 	}
 	return false, nil
 }
+
+// PodHashMatch will check the hash info in existing pod to ensure its
+// hash info matches the desired Service's hash.
+func PodHashMatch(existing, new *corev1.Pod) bool {
+	labels := existing.GetLabels()
+	newLabels := new.GetLabels()
+	if len(labels) == 0 || len(newLabels) == 0 {
+		return false
+	}
```

Comment:
Just double checking: this means that if neither `existing` nor `new` have the expected label, we consider that to mean they don't match? If so, is that worth a comment?